src/openms/source/FORMAT/MRMFeatureQCFile.cpp

```diff
@@ -28,265 +28,176 @@
+    for (const std::pair<String, Size>& h : headers) // parse the parameters
+    {
+      const String& header = h.first;
+      const Size& i = h.second;
+      boost::smatch m;
+      if (boost::regex_search(header, m, boost::regex("metaValue_(.+)_(l|u)"))) // capture the metavalue name and the boundary and save them to m[1] and m[2]
       {
-        headers[line[i]] = i;
+        setPairValue_(String(m[1]), line[i], String(m[2]), c.meta_value_qc);
       }
     }
+    c_qcs.push_back(c); // TODO: check for any condition before pushing?
   }
 
-  void MRMFeatureQCFile::parseLine_(StringList & line, std::map<String,int> & headers, 
-    std::map<String,int> & params_headers,
-    MRMFeatureQC & mrmfqc)
+  void MRMFeatureQCFile::pushValuesFromLine_(
+    const StringList& line,
+    const std::map<String, Size>& headers,
+    std::vector<MRMFeatureQC::ComponentGroupQCs>& cg_qcs
+  ) const
   {
-    // component QCs
-    MRMFeatureQC::ComponentQCs cqcs;
-    cqcs.component_name = "";
-    if (headers["component_name"] != -1)
-    {
-      cqcs.component_name = line[headers["component_name"]];
-    }
-    cqcs.retention_time_l = 0;
-    if (headers["retention_time_l"] != -1)
-    {
-      cqcs.retention_time_l = (line[headers["retention_time_l"]].empty()) ? 0.0 : std::stod(line[headers["retention_time_l"]]);
-    }
-    cqcs.retention_time_u = 0;
-    if (headers["retention_time_u"] != -1)
-    {
-      cqcs.retention_time_u = (line[headers["retention_time_u"]].empty()) ? 0.0 : std::stod(line[headers["retention_time_u"]]);
-    }    
-    cqcs.intensity_l = 0;
-    if (headers["intensity_l"] != -1)
-    {
-      cqcs.intensity_l = (line[headers["intensity_l"]].empty()) ? 0.0 : std::stod(line[headers["intensity_l"]]);
-    }
-    cqcs.intensity_u = 0;
-    if (headers["intensity_u"] != -1)
-    {
-      cqcs.intensity_u = (line[headers["intensity_u"]].empty()) ? 0.0 : std::stod(line[headers["intensity_u"]]);
-    }
-    cqcs.overall_quality_l = 0;
-    if (headers["overall_quality_l"] != -1)
-    {
-      cqcs.overall_quality_l = (line[headers["overall_quality_l"]].empty()) ? 0.0 : std::stod(line[headers["overall_quality_l"]]);
-    }
-    cqcs.overall_quality_u = 0;
-    if (headers["overall_quality_u"] != -1)
-    {
-      cqcs.overall_quality_u = (line[headers["overall_quality_u"]].empty()) ? 0.0 : std::stod(line[headers["overall_quality_u"]]);
-    }
-    // parse metaValues
-    String meta_value_key = "";
-    String lub = "";
-    std::pair<double,double> lbub {0,0};
-    for (auto const& kv : params_headers)
+    MRMFeatureQC::ComponentGroupQCs cg;
+    std::map<String, Size>::const_iterator it;
+    it = headers.find("component_group_name");
+    cg.component_group_name = it != headers.end() ? line[it->second] : "";
+    it = headers.find("retention_time_l");
+    cg.retention_time_l = it != headers.end() ? line[it->second].toDouble() : 0.0;
+    it = headers.find("retention_time_u");
```

Comment:
retention time, intensity, and quality upper bounds should be the same as for componentQCs